### PR TITLE
wrong definition of cross entropy in make_loss

### PR DESCRIPTION
**Pull request by chinakook**
Cross entropy is a cost function and lower cost is better cost. so the expression should add a minus and mean w.r.t. batch size.

### CODE REVIEW

File: src/operator/make_loss.cc [hunk 1]
```diff
@@ -37,7 +37,7 @@ The output of this function is the gradient of loss with respect to the input da
 For example, if you are a making a cross entropy loss function. Assume ``out`` is the
 predicted output and ``label`` is the true label, then the cross entropy can be defined as::
 
-  cross_entropy = label * log(out) + (1 - label) * log(1 - out)
+  cross_entropy = -mean(label * log(out) + (1 - label) * log(1 - out))
   loss = MakeLoss(cross_entropy)
```
Comment (szha) [inline src/operator/make_loss.cc]: For batch data, the [loss is already normalized by batch size in the backend](https://github.com/dmlc/mxnet/blob/master/src/operator/make_loss-inl.h#L94-L97), so `mean` is not necessary. I think you're right about the sign.

Comment (chinakook) [inline src/operator/make_loss.cc]: The normalization param of MakeLoss is set default to 'null', not 'batch'. I think that if the shapes of data and label are (N, C, H, W), the right way to calculate cross entropy, as I have tested, is to mean it on the (N, H, W) axes according to http://ufldl.stanford.edu/wiki/index.php/Softmax_Regression.

**Comment by piiswrong** [comment]
We use sum for loss to support multi-gpu training.
To sum over the non-batch axis, mean(x, axis=0, exclude=True)

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
Also, do please check out our [forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) for general "how-to" questions.
